android-core/src/main/kotlin/com/mparticle/internal/ApplicationContextWrapper.kt

```diff
@@ -0,0 +1,351 @@
+            get() {
+                var list: LinkedList<LifeCycleEvent>
+                synchronized(lifeCycleEvents) {
+                    list = LinkedList(lifeCycleEvents)
+                }
+                return list
+            }
+    }
+
+    inner class LifeCycleEvent(
+        val methodType: MethodType,
+        val activityRef: WeakReference<Activity>?,
+        val bundle: Bundle?
+    ) {
+        constructor(
+            methodType: MethodType,
+            activityRef: WeakReference<Activity>?
+        ) : this(methodType, activityRef, null)
+
+        override fun equals(o: Any?): Boolean {
+            if (o is LifeCycleEvent) {
+                val l = o
+                var matchingActivityRef = false
+                if (l.activityRef == null && activityRef == null) {
+                    matchingActivityRef = true
+                } else if (l.activityRef != null && activityRef != null) {
+                    matchingActivityRef = l.activityRef.get() === activityRef.get()
+                }
+                return matchingActivityRef && l.methodType == methodType && l.bundle == bundle
+            }
+            return false
+        }
+    }
+
+    internal inner class ReplayLifecycleCallbacksRunnable(var callback: ActivityLifecycleCallbacks?) :
+        Runnable {
+        @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+        override fun run() {
+            if (callback != null && mActivityLifecycleCallbackRecorder != null && isReplayActivityLifecycle) {
+                val reference = if (MParticle.getInstance()!!
+                    .Internal().kitManager == null
+                ) {
+                    null
+                } else {
+                    MParticle.getInstance()!!
+                        .Internal().kitManager.currentActivity
+                }
+                if (reference != null) {
+                    val currentActivity = reference.get()
+                    if (currentActivity != null) {
+                        val recordedLifecycleList: LinkedList<LifeCycleEvent> =
+                            mActivityLifecycleCallbackRecorder!!.recordedLifecycleListCopy
+                        while (recordedLifecycleList.size > 0) {
+                            val lifeCycleEvent = recordedLifecycleList.removeFirst()
+                            if (lifeCycleEvent.activityRef != null) {
+                                val recordedActivity = lifeCycleEvent.activityRef.get()
+                                if (recordedActivity != null) {
+                                    if (recordedActivity === currentActivity) {
+                                        when (lifeCycleEvent.methodType) {
+                                            MethodType.ON_CREATED -> {
+                                                Logger.debug("Forwarding OnCreate")
+                                                callback!!.onActivityCreated(
```

Comment:
Is there any way we can write this without force unwrapping? Same for all uses of `!!` below.